parastell/utils.py

```diff
@@ -93,14 +94,25 @@ def expand_list(list, num):
     avg_diff = extent / (num - 1)
 
     for entry, next_entry in zip(list[:-1], list[1:]):
-        num_new_entries = int(round((next_entry - entry) / avg_diff))
-
-        # Don't append the last entry in the created linspace to avoid adding
-        # it twice when the next created linspace is appended
-        list_exp = np.append(
-            list_exp,
-            np.linspace(entry, next_entry, num=num_new_entries + 1)[:-1],
-        )
+        # Only add entries to current block if difference between entry and
+        # next_entry is greater than desired average
+        num_new_entries = (next_entry - entry) / avg_diff
+        if num_new_entries >= 1.0:
+            num_new_entries = int(round(num_new_entries))
+        else:
+            num_new_entries = 0
```

Comment:
This seems slightly cleaner to me (?) and matches your comment a little better, perhaps
```suggestion
        num_new_entries = 0;
        
        if next_entry - entry > avg_diff:
            num_new_entries = int(round(next_entry - entry / avg_diff))
```